lib/ruby_smb/server.rb

```diff
@@ -58,6 +58,14 @@ def add_share(share_provider)
       @shares[share_provider.name] = share_provider
     end
 
+    def remove_share(share_provider)
+      share_provider = share_provder.name if share_provider.is_a?(RubySMB::Server::Share::Provider::Base)
```

Comment:
Two things.....
One, this looks like a typo?
```suggestion
      share_provider = share_proivder.name if share_provider.is_a?(RubySMB::Server::Share::Provider::Base)
```

Two, this looks like we're taking an element of an object and overwriting that object with the object's element?
On second thought, I see.  I'm guessing that `@shares.delete()` takes a string that is the `share_provider.name` value, but this line will overwrite the parameter to the `share_provider.name` value if the user passes in the `share_provider` object.
Could we maybe add a different variable name for this like `share_provider_name`, or even `share_provider_to_delete`?  Maybe even just a comment that this can take either the name or the `share_provider` object?

Reply:
1. Yes, that's a typo and I'll fix it.
2. I'm confused as to what you're saying. Basically `share_provider` needs to be the share name to remove. This line only serves to obtain that value from the `#name` attribute if the `share_provider` is an instance of `RubySMB::Server::Share::Provider::Base`, otherwise it needs to be a string. I'll add some docs to clarify that.

Reply:
Per #2- It just struck me as odd we were renaming an object to an objects attribute.  I think the crux was that you're leveraging weak typing here as a feature to create a method that is designed to take multiple parameter types and deal with the properly, but even after years of Ruby and Python, it came as a surprise to me.  I think just a quick one-liner to explain that was your goal, or to have something documenting the multitype support for the method definition.

Reply:
I added some YARD docs for the parameter that call out it accepts either a String or `Share::Provider::Base` instance. Let me know if you think I can make it more clear.